public/components/create_report/create_report.tsx

```diff
@@ -38,14 +38,34 @@ import {
     EuiRadioGroup,
   } from '@elastic/eui';
 import { htmlIdGenerator } from '@elastic/eui/lib/services';
-import moment from 'moment';
+import moment, { Moment } from 'moment';
 import { ReportSettings } from './report_settings/report_settings';
 import { ReportDelivery } from './delivery/delivery';
 import { ReportSchedule } from './schedule/schedule';
 
 const idPrefix = htmlIdGenerator()();
 
-export class CreateReport extends React.Component {
+interface RouterHomeProps {
+  httpClient?: any
+}
+
+interface CreateReportState {
+  reportSettingsRadioIdSelected: string,
+  reportSettingsSetRadioIdSelected: string,
+  reportSettingsDashboard: string,
+  deliveryEmailSubject: string,
+  deliveryEmailBody: string,
+  scheduleRadioFutureDateSelected: boolean,
+  scheduleRadioRecurringSelected: boolean,
+  scheduleUTCOffset: number,
+  scheduleRecurringFrequency: string,
+  scheduleRecurringDailyTime: string, 
+  scheduleRecurringUTCOffset: number,
+  scheduleRecurringWeeklyDayOfWeek: string,
+  scheduleRecurringStartDate: Moment
+}
+
+export class CreateReport extends React.Component<RouterHomeProps, CreateReportState> {
```

Comment:
np: Use functional components and hooks make things much more easy to write no more overhead on classes. 

* https://reactjs.org/docs/hooks-intro.html 
You can take look at AD plugin as well.